crates/rpc/src/routes.rs

```diff
@@ -0,0 +1,25 @@
+use std::sync::Arc;
+
+use warp::{reply::Reply, Filter};
+
+use crate::{handlers::genesis::get_genesis, types::genesis::Genesis};
+
+/// Creates and returns all possible routes.
+pub fn get_routes(
+    ctx: Arc<Genesis>,
+) -> impl Filter<Extract = impl Reply, Error = warp::Rejection> + Clone {
+    let eth_base = warp::path("eth")
+        .and(warp::path("v1"))
+        .and(warp::path("beacon"));
+
+    let genesis_clone: Arc<Genesis> = Arc::clone(&ctx);
+    let genesis = eth_base
+        .and(warp::path("genesis"))
+        .and(warp::get())
+        .and_then(move || get_genesis(genesis_clone.clone()))
+        .with(warp::log("genesis"));
+
+    #[allow(clippy::let_and_return)]
+    let routes = genesis;
+    routes
+}
```

Comment:
```suggestion
/// Creates and returns all possible routes.
pub fn get_routes(
    network_spec: Arc<NetworkSpec>,
) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
    let eth_base = path("eth")
        .and(path("v1"))
        .and(path("beacon"));

    let genesis = eth_base
        .and(warp::path("genesis"))
        .and(warp::get())
        .and_then(move || get_genesis(network_spec.clone()))
        .with(warp::log("genesis"));

    genesis
}
```

I don't think we should be using `#[allow(clippy::let_and_return)]`

Reply:
I was storing it in a another variable since eventually the other filter endpoints would add up with the genesis filter to create a final route filter